e2e/policy/policy_test.go

```diff
@@ -0,0 +1,175 @@
+
+		// parse the original resources
+		resourceBytes, err := os.ReadFile(path.Join(ct.WorkDir, "resources.yaml"))
+		require.NoError(err)
+		r, err := kubeapi.UnmarshalUnstructuredK8SResource(resourceBytes)
+		require.NoError(err)
+
+		// remove everything from the openssl-frontend
+		newResources := r[:0]
+		for _, resource := range r {
+			name := resource.GetName()
+			require.NoError(err)
+			if name == opensslFrontend || name == "port-forwarder-"+opensslFrontend {
+				continue
+			}
+			newResources = append(newResources, resource)
+		}
+
+		// write the new resources yaml
+		r = newResources
+		resourceBytes, err = kuberesource.EncodeUnstructured(r)
+		require.NoError(err)
+		require.NoError(os.WriteFile(path.Join(ct.WorkDir, "resources.yaml"), resourceBytes, 0o644))
+
+		// set the new manifest
+		ct.Set(t)
+
+		// restart the deployments
+		require.NoError(c.Restart(ctx, kubeclient.Deployment{}, ct.Namespace, opensslFrontend)) // not waiting since it would fail
+		require.NoError(c.Restart(ctx, kubeclient.Deployment{}, ct.Namespace, opensslBackend))
+		require.NoError(c.WaitFor(ctx, kubeclient.Deployment{}, ct.Namespace, opensslBackend))
+
+		// wait a bit to let the attestation failure counter go up
+		time.Sleep(5 * time.Second)
+
+		newFailures := getFailures(ctx, t, ct)
+		t.Log("New failures:", newFailures)
+		// errors should happen
+		require.Greater(newFailures, initialFailures)
+	})
+}
+
+func TestMain(m *testing.M) {
+	flag.StringVar(&imageReplacementsFile, "image-replacements", "", "path to image replacements file")
+	flag.StringVar(&namespaceFile, "namespace-file", "", "file to store the namespace in")
+	flag.BoolVar(&skipUndeploy, "skip-undeploy", false, "skip undeploy step in the test")
+	flag.Parse()
+
+	os.Exit(m.Run())
+}
+
+func parsePrometheus(input string) (map[string]*dto.MetricFamily, error) {
+	var parser expfmt.TextParser
+	return parser.TextToMetricFamilies(strings.NewReader(input))
+}
+
+func getFailures(ctx context.Context, t *testing.T, ct *contrasttest.ContrastTest) int {
+	require := require.New(t)
+
+	coordPods, err := ct.Kubeclient.PodsFromOwner(ctx, ct.Namespace, "StatefulSet", coordinator)
+	require.NoError(err)
+	require.NotEmpty(coordPods, "pod not found: %s/%s", ct.Namespace, coordinator)
+	coordIP := coordPods[0].Status.PodIP
+	backendPods, err := ct.Kubeclient.PodsFromDeployment(ctx, ct.Namespace, opensslBackend)
+	require.NoError(err)
+	require.NotEmpty(backendPods, "pod not found: %s/%s", ct.Namespace, opensslBackend)
+	metricsString, _, err := ct.Kubeclient.Exec(ctx, ct.Namespace, backendPods[0].Name, []string{"curl", coordIP + ":9102/metrics"})
+	require.NoError(err)
+	metrics, err := parsePrometheus(metricsString)
+	require.NoError(err)
+	var failures int
+	for k, v := range metrics {
+		if k == "contrast_meshapi_attestation_failures" {
+			failures = int(v.GetMetric()[0].GetCounter().GetValue())
+		}
+	}
```

Comment:
You could add a `t.Error` here to capture the unexpected condition of an absent failure metric.